resqpy/property.py

```diff
@@ -3039,6 +3039,332 @@ def basic_static_property_uuids_dict(self, realization = None, share_perm_parts
+
+      arguments:
+         parent_model (model.Model): the model to which the property belongs
+         uuid (uuid.UUID, optional): required if initialising from an existing RESQML property object
+         title (str, optional): the citation title to use for the property; ignored if uuid is present
+         support_uuid (uuid.UUID, optional): identifies the supporting representation for the property;
+            ignored if uuid is present
+         extra_metadata (dict, optional): if present, the dictionary items are added as extra metadata
+
+      returns:
+         new resqpy Property object
+      """
+
+      self.collection = PropertyCollection()
+      super().__init__(model = parent_model, uuid = uuid, title = title, extra_metadata = extra_metadata)
+      if support_uuid is not None:
+         if self.collection.support_uuid is None:
+            self.collection.set_support(model = parent_model, support_uuid = support_uuid)
+         else:
+            assert bu.matching_uuids(support_uuid, self.collection.support_uuid)
+
+   def load_from_xml(self):
+      """Populates this property object from xml; not usually called directly."""
+
+      part = self.part
+      assert part is not None
+      if self.collection is None: self.collection = PropertyCollection()
+      if self.collection.model is None: self.collection.model = self.model
+      self.collection.add_part_to_dict(part)
+      self.collection.has_single_property_kind_flag = True
+      self.collection.has_single_indexable_element_flag = True
+      self.collection.has_multiple_realizations_flag = False
+      assert self.collection.number_of_parts() == 1
+
+   @classmethod
+   def from_singleton_collection(cls, property_collection: PropertyCollection):
+      """Populates this (empty) property from a PropertyCollection containing just one part.
+
+      arguments:
+         property_collection (PropertyCollection): the singleton collection from which to populate this Property
+      """
+
+      # Validate
+      assert property_collection.model is not None
+      assert property_collection is not None
+      assert property_collection.number_of_parts() == 1
+
+      # Instantiate the object i.e. call the class __init__ method
+      part = property_collection.parts()[0]
+      prop = cls(
+         parent_model=property_collection.model,
+         uuid=property_collection.uuid_for_part(part),
+         title=property_collection.citation_title_for_part(part),
+         extra_metadata=property_collection.extra_metadata_for_part(part)
+      )
+
+      # Edit properties of parent collection
+      prop.collection.has_single_property_kind_flag = True
+      prop.collection.has_single_indexable_element_flag = True
+      prop.collection.has_multiple_realizations_flag = False
+
+      return prop
+
+   def array_ref(self, dtype = None, masked = False, exclude_null = False):
```

Comment:
A tentative suggestion here: to be more "pythonic" and fit in with the look-and-feel of other scientific computing libraries, this method could be named `values()` .

If I didn't read the docstring, I might guess that `array_ref` was some kind of reference to the HDF5 location; whereas it actually seems to be the most important method that returns the actual property data

Reply:
I'd rather postpone switching to "values" as we would have to change several method names throughout resqpy (and go through a deprecation cycle for the array_ref or similar names). I'll create a separate issue for now.